lib/vrl/stdlib/src/reverse.rs

```diff
@@ -0,0 +1,78 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Reverse;
+
+impl Function for Reverse {
+    fn identifier(&self) -> &'static str {
+        "reverse"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::ARRAY,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "reverse array",
+            source: r#"reverse([0, 1])"#,
+            result: Ok("[1, 0]"),
+        }]
+    }
+
+    fn compile(&self, _state: &state::Compiler, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(Box::new(ReverseFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ReverseFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ReverseFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let mut value = self.value.resolve(ctx)?.try_array()?;
+
+        value.reverse();
+
+        Ok(value.into())
+    }
+
+    fn type_def(&self, _state: &state::Compiler) -> TypeDef {
+        TypeDef::new().array_mapped::<(), Kind>(map! { (): Kind::all() })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    test_function![
+        reverse => Reverse;
+
+        both_arrays_empty {
+            args: func_args![value: value!([]), items: value!([])],
+            want: Ok(value!([])),
+            tdef: TypeDef::new().array_mapped::<(), Kind>(map! { (): Kind::all() }),
+        }
+
+        one_array_empty {
+            args: func_args![value: value!([1, 2, 3])],
+            want: Ok(value!([3, 2, 1])),
+            tdef: TypeDef::new().array_mapped::<(), Kind>(map! { (): Kind::all() }),
```

Comment:
Theoretically this could have been:

```rust
            tdef: TypeDef::new().array_mapped::<(), Kind>(map! { (): Kind::Integer }),
```

However, the test here setups up the typedef for the array by defining each index as an integer:

```
1 -> Kind::Integer
2 -> Kind::Integer
3 -> Kind::Integer
```

So, the typedef here should be as it already is since theoretically VRL can't guarantee that at runtime there will not be a fourth element in the array that is a String.